### DISCUSSION TITLE
Pylint config file

### DISCUSSION THREAD

**Discussion by simonf-dev** [akaihola/darker#398] [Q&A] [answered]
Hello, I'm trying to pass Darker my custom setup for pylint, but it seems that Darker doesn't care about my settings at all:
My settings pyproject.toml:
[tool.pylint.message_control]
disable="no-self-use"

Command that I use:

darker --config pyproject.toml --rev master -L pylint . 

Am I doing something wrong, or it's impossible to pass pylint my settings right now?

**Comment by akaihola** [comment, marked as answer]
Hi @simonf-dev, thanks for your question!

I've successfully configured Pylint by using either a `.pylint` or `pylintrc` file at the root of the repository. To suppress messages, you can use a section like in this example:
```ini
[MESSAGES CONTROL]
disable=wrong-import-order
disable=C0103,C0111,C0302,R0903,W0142
```

Darker uses the file pointed to by the `--config` parameter to configure its own behavior and to pass certain options on to Black and Isort. It doesn't attempt to affect linters' settings in any way.

The `-L`/`--lint` parameter treats its value (`pylint` in your example) as just a command line to invoke. Darker has no understanding of any particular linters you may run using that option. So configuring of any linters must be done with the linter's own mechanisms.

You _can_ provide command line arguments to linters by quoting a command line, e.g.:
```bash
darker --lint "pylint --disable=no-self-use" .
```

I would like to make sure the documentation of Darker is clear on this topic as well. If you feel like the documentation has misled you, could you point to the section to improve? Thanks!

**Reply by simonf-dev** [reply]
Thank you for your help! It works properly. Possible improvements in docs:
1. '--help' command can specify that config option is only for black/isort/darker things
2. '--help' command can specify that -L option config is taken automatically from root directory as normally and you can pass arguments as normally
3. Maybe you can also mention these things in your github page docs, I was really confused that black settings work properly with --config option but pylint doesn't

**Reply by akaihola** [reply]
Thanks @simonf-dev, these suggestions are now queued up in issue #399.
